src/main/angular/app/services/graph-processor.service.ts

```diff
@@ -0,0 +1,275 @@
+import {Injectable, SimpleChanges} from '@angular/core';
+import {ProjectGraphData} from '../../model/project-graph-data';
+import {Recommendation} from '../../model/recommendation';
+import {Project} from '../../model/project';
+import {SimpleChange} from '@angular/core';
+import {DateUtilitiesService} from './date-utilities.service';
+import {GraphProperties, Columns, Row} from '../../model/types';
+import {DataService} from './data.service';
+
+/** Provides methods to convert data to the format used by Google Charts. */
+@Injectable()
+export class GraphProcessorService {
+  constructor(private dateUtilities: DateUtilitiesService) {}
+
+  /** Initialize the chart properties with empty data. */
+  initProperties(): GraphProperties {
+    const options: google.visualization.LineChartOptions = {
+      animation: {
+        duration: 500,
+        easing: 'out',
+        startup: true,
+      },
+      legend: {position: 'none'},
+      hAxis: {
+        gridlines: {
+          color: 'white',
+        },
+        format: 'M/d',
+        minTextSpacing: 100,
+      },
+      vAxis: {
+        minorGridlines: {
+          color: 'white',
+        },
+      },
+      series: {},
+    };
+    return {
+      options: options,
+      graphData: [],
+      columns: ['Time'],
+      type: 'LineChart',
+      title: 'Empty Graph',
+      width: 960,
+      height: 600,
+    };
+  }
+
+  /** Process the given changes and adjust from the graph properties as necessary. */
+  async processChanges(
+    changes: SimpleChanges,
+    properties: GraphProperties,
+    httpService: DataService
+  ): Promise<void> {
+    const additionsDeletions = this.getAdditionsDeletions(changes.projects);
+    const promises: Promise<unknown>[] = [];
+
+    additionsDeletions.added.forEach(addition =>
+      promises.push(
+        httpService
+          .getProjectGraphData(addition.projectId)
+          .then(data => this.addToGraph(properties, data, addition))
+      )
+    );
+    additionsDeletions.removed.forEach(removal =>
+      this.removeFromGraph(properties, removal)
+    );
+    return Promise.all(promises).then(() => {
+      properties.title = 'IAM Bindings';
+      return undefined;
+    });
+  }
+
+  /** Adds the given project to the graph. */
+  private addToGraph(
+    properties: GraphProperties,
+    data: ProjectGraphData,
+    project: Project
+  ) {
+    const seriesNum: number = (properties.columns.length - 1) / 3;
```

Comment:
avoid abbreviations, we have autocomplete in our IDEs so it's hassle free to have fully spelled out variable names